docs/section19/19-2.md

```diff
@@ -0,0 +1,74 @@
+
+**Function Calls:**  
+1. is_hvac_sys_fluid_loop_attached_to_boiler()
+2. is_hvac_sys_cooling_type_fluid_loop()
+3. is_hvac_sys_heating_type_fluid_loop()
+4. is_hvac_sys_fluid_loop_attached_to_chiller()
+5. is_hvac_sys_preheating_type_fluid_loop()  
+6. is_hvac_sys_preheat_fluid_loop_attached_to_boiler()  
+7. are_all_terminal_heating_loops_attached_to_boiler()
+
+
+
+## Rule Logic:   
+- Cycle through each HVAC system in the B-RMR to create a list of HW loops for space heating served by boiler(s) and a list of CHW loop for space cooling served by chiller(s) and a list of condenser water loops: `for each hvac in B_RMR...HeatingVentilatingAirConditioningSystem:`  
+    - Check if the hvac system heating coil is a fluid loop: `if is_hvac_sys_heating_type_fluid_loop(B_RMR, hvac.id)  == TRUE:`  
+        - Check if the fluid loop is attached to a boiler, if yes then add to list of HW fluid loops: `if is_hvac_sys_fluid_loop_attached_to_boiler(B_RMR, hvac.id) ==  TRUE: HW_fluid_loop_list =  HW_fluid_loop_list.append(hvac.heating_system.hot_water_loop)` 
+    - Check if the hvac system preheating coil is a fluid loop: `if is_hvac_sys_preheating_type_fluid_loop(B_RMR, hvac.id)  == TRUE:`  
+        - Check if the fluid loop is attached to a boiler, if yes then add to list of HW fluid loops: `if is_hvac_sys_preheat_fluid_loop_attached_to_boiler(B_RMR, hvac.id) ==  TRUE: HW_fluid_loop_list =  HW_fluid_loop_list.append(hvac.heating_system.hot_water_loop)` 
+    - Check if the hvac system cooling coil is a fluid loop: `if is_hvac_sys_cooling_type_fluid_loop(B_RMR, hvac.id)  == TRUE:`  
+        - Check if the fluid loop is attached to a chiller, if yes then add to list of CHW fluid loops and CW loops: `if is_hvac_sys_fluid_loop_attached_to_chiller(B_RMR, hvac.id) ==  TRUE:`   
+            - Add CHW fluid loop to chilled water loop list: `CHW_fluid_loop_list =  CHW_fluid_loop_list.append(hvac.cooling_system.chilled_water_loop)`    
+            - Add condenser water fluid loop to condenser water loop list: `CW_fluid_loop_list =  CW_fluid_loop_list.append(hvac.cooling_system.condenser_water_loop)`  
+            - Add the child loops to the CHW loop list: `For CHW_child_loop in hvac.cooling_system.chilled_water_Loop.child_loops:`  
+                - Add to the list of chilled water loops: `CHW_fluid_loop_list =  CHW_fluid_loop_list.append(CHW_child_loop)`             
+- Cycle through each terminal unit in the B-RMR and create list of HW loops: `for terminal in B_RMR...Terminal:`  
+    - Reset list with the one terminal unit: `terminal_list = []` 
+    - Check if heating source is hot water: `if terminal.heating_source == "HOT_WATER":`  
+        - Create a list with the one terminal unit: `terminal_list = [terminal.id]`  
+        - Check if the loop is attached to a boiler: `if are_all_terminal_heating_loops_attached_to_boiler(B_RMR,terminal_list) ==  TRUE:`  
```

Comment:
@mkarpman We are only sending that function one terminal unit. The only reason I create a list with one value is because that function expects a list but there is only one terminal being sent to the function. Does this make your comment obsolete?

Reply:
instead of creating terminal_list, you could amend line 45 to read:

`if are_all_terminal_heating_loops_attached_to_boiler(B_RMR,[terminal.id]) ==  TRUE:`

this makes it a little more obvious.

Also, the check on line 46 is redundant, lines 48-50 already eliminate duplicates